src/crimson/net/SocketMessenger.cc

```diff
@@ -154,15 +155,18 @@ SocketMessenger::connect(const entity_addr_t& peer_addr, const entity_name_t& pe
   assert(seastar::this_shard_id() == master_sid);
 
   // make sure we connect to a valid peer_addr
-  ceph_assert(peer_addr.is_legacy() || peer_addr.is_msgr2());
+  if (!peer_addr.is_msgr2()) {
+    logger().warn("ProtocolV1 is no longer supported");
+    return nullptr;
```

Comment:
I think `ceph_abort()` here is better. The current crimson OSD code doesn't expect or handle `nullptr` returned by `Messenger::connect()`. And if OSD need to deal with it, it should be done before calling `Messenger::connect()`.

Reply:
Dropped.